modules/RTC/TPCUtils.js

```diff
@@ -80,37 +158,103 @@ export class TPCUtils {
      * scaleResolutionDownBy - the factor by which the encoding is scaled down from the
      *  original resolution of the captured video.
      *
-     *  @param {VideoType} videoType
+     * @param {VideoType} videoType
+     * @param {String} codec
      */
-    _getVideoStreamEncodings(videoType) {
-        const maxVideoBitrate = videoType === VideoType.DESKTOP && this.encodingBitrates.ssHigh
-            ? this.encodingBitrates.ssHigh : this.encodingBitrates.high;
+    _getVideoStreamEncodings(videoType, codec) {
+        const codecBitrates = this.codecSettings[codec].maxBitratesVideo;
+        const maxVideoBitrate = videoType === VideoType.DESKTOP
+            ? codecBitrates.ssHigh : codecBitrates.high;
 
         // The SSRCs on older versions of Firefox are reversed in SDP, i.e., they have resolution order of 1:2:4 as
         // opposed to Chromium and other browsers. This has been reverted in Firefox 117 as part of the below commit.
         // https://hg.mozilla.org/mozilla-central/rev/b0348f1f8d7197fb87158ba74542d28d46133997
         const reversedEncodings = browser.isFirefox() && browser.isVersionLessThan(117);
 
-        return [
+        const standardSimulcastEncodings = [
             {
                 active: this.pc.videoTransferActive,
-                maxBitrate: reversedEncodings ? maxVideoBitrate : this.encodingBitrates.low,
+                maxBitrate: reversedEncodings ? maxVideoBitrate : codecBitrates.low,
                 rid: SIM_LAYER_1_RID,
                 scaleResolutionDownBy: reversedEncodings ? HD_SCALE_FACTOR : LD_SCALE_FACTOR
             },
             {
                 active: this.pc.videoTransferActive,
-                maxBitrate: this.encodingBitrates.standard,
+                maxBitrate: codecBitrates.standard,
                 rid: SIM_LAYER_2_RID,
                 scaleResolutionDownBy: SD_SCALE_FACTOR
             },
             {
                 active: this.pc.videoTransferActive,
-                maxBitrate: reversedEncodings ? this.encodingBitrates.low : maxVideoBitrate,
+                maxBitrate: reversedEncodings ? codecBitrates.low : maxVideoBitrate,
                 rid: SIM_LAYER_3_RID,
                 scaleResolutionDownBy: reversedEncodings ? LD_SCALE_FACTOR : HD_SCALE_FACTOR
             }
         ];
+
+        if (this.codecSettings[codec].scalabilityModeEnabled) {
+            // Configure all 3 encodings when simulcast is requested through config.js for AV1 and VP9 and for H.264
+            // always since that is the only supported mode when DD header extension is negotiated for H.264.
```

Comment:
I see no check for the DD header extension negotiation here. What if someone with a new LJM connects to a bridge without that? Is that possible?

Reply:
Jonathan just discovered that DD header extension is negotiated whenever AV1 is offered by Jicofo. I am working on AV1-DD suppression logic (for VP8 and VP9) so I can add a check for that here. We decided not to merge this PR without the suppression logic since it affects scalability for VP8.